modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -191,17 +192,26 @@ public void validate(BaseModel<?> baseModel, long objectDefinitionId)
 			return;
 		}
 
-		HashMapBuilder.HashMapWrapper<String, Object> hashMapWrapper =
-			HashMapBuilder.<String, Object>putAll(
-				baseModel.getModelAttributes());
+		Map<String, Object> values = new HashMap<>();
 
 		if (baseModel instanceof ObjectEntry) {
-			Map<String, Serializable> values =
-				_objectEntryLocalService.getValues((ObjectEntry)baseModel);
-
-			if (values != null) {
-				hashMapWrapper.putAll(values);
-			}
+			values = HashMapBuilder.<String, Object>putAll(
+				baseModel.getModelAttributes()
+			).putAll(
+				_objectEntryLocalService.getValues((ObjectEntry)baseModel)
+			).build();
+		}
+		else {
+			values = HashMapBuilder.<String, Object>putAll(
+				baseModel.getModelAttributes()
+			).putAll(
+				_objectEntryLocalService.getSystemObjectExtendedProperties(
+					_objectDefinitionPersistence.fetchByPrimaryKey(
+						objectDefinitionId),
+					GetterUtil.getLong(baseModel.getPrimaryKeyObj()))
+			).putAll(
+				EntityExtensionThreadLocal.getExtendedProperties()
```

Comment:
Do you think that `getExtendedProperties` coming last will be a problem?

Reply:
Same as above